lib/block_view_dao_coin_limit_order.go

```diff
@@ -37,37 +38,135 @@ func (bav *UtxoView) _connectDAOCoinLimitOrder(
 		// public key so there is no need to verify anything further.
 	}
 
+	// ----- Begin custom validations
+
+	// Validate TransactorPKID exists.
+	transactorPKID := bav.GetPKIDForPublicKey(txn.PublicKey).PKID
+
+	if transactorPKID == nil {
+		return 0, 0, nil, RuleErrorDAOCoinLimitOrderInvalidTransactorPKID
+	}
+
+	// Validate DenominatedCoinType is one of our supported enum values and is always $DESO for now.
+	switch txMeta.DenominatedCoinType {
+	case DAOCoinLimitOrderEntryDenominatedCoinTypeDESO:
+		break
+	case DAOCoinLimitOrderEntryDenominatedCoinTypeDAOCoin:
+		return 0, 0, nil, RuleErrorDAOCoinLimitOrderUnsupportedDenominatedCoinType
+	default:
+		return 0, 0, nil, RuleErrorDAOCoinLimitOrderUnsupportedDenominatedCoinType
+	}
+
+	// If denominated in $DESO, validate DenominatedCoinCreatorPKID is all zeroes.
+	if txMeta.DenominatedCoinType == DAOCoinLimitOrderEntryDenominatedCoinTypeDESO {
+		if !reflect.DeepEqual(ZeroPKID, txMeta.DenominatedCoinCreatorPKID) {
```

Comment:
This will always be false since `ZeroPKID` is a direct reference to a `PKID` and `txMeta.DenominatedCoinCreatorPKID` is a pointer.
```suggestion
		if !reflect.DeepEqual(ZeroPKID, *txMeta.DenominatedCoinCreatorPKID) {
```